components/src/components/event-list/event-list.tsx

```diff
@@ -0,0 +1,193 @@
+import { Component, Host, h, Prop, State } from "@stencil/core";
+import { PulumiEvent, PulumiRawEvent, PulumiRawWebinar } from "../../util/types/events";
+import { classNames, shuffleArray } from "../../util/util";
+
+// Event types
+type EventFilterType = "featured" | "upcoming" | "on-demand" | "pulumitv";
+type EventFilterChoiceText = "All" | "Featured" | "Upcoming Sessions" | "Videos" | "PulumiTV";
+type EventFilter = "all" | EventFilterType;
+
+interface EventFilterItem {
+    text: EventFilterChoiceText;
+    key: EventFilter;
+    icon: string;
+}
+
+// parseEventsString takes an JSON string of the data for an "event"
+// and returns a typed PulumiEvent that we use to create the listing.
+function parseEventsString(eventsString: string): PulumiEvent[] {
+    const rawEvents: PulumiRawEvent[] = JSON.parse(eventsString);
+    return rawEvents.map((event) => {
+        return {
+            archetype: "event",
+            title: event.title,
+            description: event?.event?.description || "",
+            startDate: new Date(event?.event?.start_date || ""),
+            registrationUrl: event?.event?.registration_url || "",
+            external: true,
+            type: [ "upcoming "],
+            imageSrc: "/images/webinar/pulumi_tech_talk.jpg"
+        }
+    });
+}
+
+// parseWebinarsString takes an JSON string of the data for an "webinar"
+// and returns a typed PulumiEvent that we use to create the listing.
+function parseWebinarsString(webinarString: string): PulumiEvent[] {
+    const rawWebinars: PulumiRawWebinar[] = JSON.parse(webinarString);
+    const result: PulumiEvent[] = [];
+
+    for (let i = 0; i < rawWebinars.length; i++) {
+        const webinar = rawWebinars[i];
+
+        const types: EventFilterType[] = [];
+
+        // Add the different types of webinars so we can attach the appropriate icon
+        // to the listing.
+        if (new Date().getTime() <= new Date(webinar.main.sortable_date).getTime()) {
+            types.push("upcoming");
+        }
+
+        if (webinar.pre_recorded) {
+            types.push("on-demand");
+        }
+
+        if (webinar.pulumi_tv) {
+            types.push("pulumitv");
+        }
+
+        if (webinar.featured) {
+            types.push("featured");
+        }
+
+        // Do not list the webinar if there is no available type.
+        if (!types.length) {
+            continue;
+        }
+
+        const pulumiEvent: PulumiEvent = {
+            archetype: "webinar",
+            title: webinar.title,
+            description: webinar.meta_desc,
+            startDate: new Date(webinar.main.sortable_date),
+            registrationUrl: webinar.url_slug,
+            external: webinar.external,
+            type: types,
+            imageSrc: webinar.preview_image,
+        };
+
+        result.push(pulumiEvent);
+    }
+
+    return result;
+}
+
+@Component({
```

Comment:
I suspect this component is doing way more than it should. When you think about what this component is for, it's basically this:

* Expose a list of things to filter by
* Handle clicks and filter accordingly

All it needs to be able to do that is show those filters and handle their clicks, and to do _that,_ all it needs to do is set a CSS class on itself that says what the selected thing is, and then CSS can take over by hiding (`display: none`, or Tailwind `hidden`) all of the list items that shouldn't be visible. I _think_ that's about it. The rest is Hugo-specific implementation stuff that doesn't need to go into either of these components really. 

So I think you could make this much simpler by just having the Hugo template use each webinar/event's `.Params` metadata to figure out what to pass into each `pulumi-event-list-item` to ensure each item gets a class that corresponds with the kind of item it is, and scaling the list component back to deal only with filtering. I can imagine the API looking something like this, then:

```
<pulumi-event-list selected="all">
    <pulumi-event-list-item title="Something" description="Something" type="webinar" date="2020-10-31" href="/resources/blah"></pulumi-event-list-item>
    <pulumi-event-list-item title="Something else" description="Something else" type="event" date="2020-10-31" href="/resources/blah"></pulumi-event-list-item>
</pulumi-event-list>
```